aws_lambda_builders/utils.py

```diff
@@ -68,15 +68,27 @@ def copytree(source, destination, ignore=None):
 # Copyright 2019 by the Python Software Foundation
 
 
-def which(cmd, mode=os.F_OK | os.X_OK, path=None):  # pragma: no cover
-    """Given a command, mode, and a PATH string, return the paths which
-    conforms to the given mode on the PATH, or None if there is no such
-    file.
-    `mode` defaults to os.F_OK | os.X_OK. `path` defaults to the result
-    of os.environ.get("PATH"), or can be overridden with a custom search
-    path.
+def which(cmd, mode=os.F_OK | os.X_OK, executable_search_paths=None):  # pragma: no cover
```

Comment:
Can you update Path Resolver design doc to explain what this is? https://github.com/awslabs/aws-lambda-builders/blob/develop/designs/Path_Resolver_DESIGN.md

Reply:
Sure, shall I do that in another PR?

Reply:
Works